paymentapi/src/main/java/fi/hel/verkkokauppa/payment/model/Payment.java

```diff
@@ -63,6 +63,9 @@ public class Payment {
 	@Field(type = FieldType.Text)
 	String paytrailTransactionId;
 
+	@Field(type = FieldType.Boolean)
+	boolean shopInShopPayment;
```

Comment:
There is no way to determine in the return callback methods if a payment is shopInShop payment or not - this leads to malformed signature calculation because it requires the right secret key (**normal merchant flow** -> merchant specific secret key | **shop-in-shop flow** -> system specified aggregate secret key). This new field will be set on payment creation to indicate what type of paytrail flow is used for the specific payment.

Reply:
Could you add this text to code aswell in some kind of FYI information?